src/draw/lv_draw_vector.h

```diff
@@ -0,0 +1,541 @@
+                                              lv_vector_gradient_spread_t spread);
+/**
+ * Set stroke radial gradient for descriptor
+ * @param dsc              pointer to a vector graphic descriptor
+ * @param grad             pointer to a `lv_grad_dsc_t` variable
+ * @param cx               the x for center of the circle
+ * @param cy               the y for center of the circle
+ * @param radius           the radius for circle
+ * @param spread           the gradient spread to be set in lv_vector_gradient_spread_t format
+ */
+void lv_vector_dsc_set_stroke_radial_gradient(lv_vector_dsc_t * dsc, const lv_grad_dsc_t * grad, float cx, float cy,
+                                              float radius, lv_vector_gradient_spread_t spread);
+/**
+ * Set a matrix to current stroke transformation matrix
+ * @param dsc              pointer to a vector graphic descriptor
+ * @param matrix           pointer to a matrix
+ */
+void lv_vector_dsc_set_stroke_transform(lv_vector_dsc_t * dsc, const lv_matrix_t * matrix);
+
+/**
+ * Set current transformation matrix to identity matrix
+ * @param dsc           pointer to a vector graphic descriptor
+ */
+void lv_vector_dsc_identity(lv_vector_dsc_t * dsc);
+
+/**
+ * Change the scale factor of current transformation matrix
+ * @param dsc           pointer to a vector graphic descriptor
+ * @param scale_x       the scale factor for the X direction
+ * @param scale_y       the scale factor for the Y direction
+ */
+void lv_vector_dsc_scale(lv_vector_dsc_t * dsc, float scale_x, float scale_y);
+
+/**
+ * Rotate current transformation matrix with origin
+ * @param dsc           pointer to a vector graphic descriptor
+ * @param degree        angle to rotate
+ */
+void lv_vector_dsc_rotate(lv_vector_dsc_t * dsc, float degree);
+
+/**
+ * Translate current transformation matrix to new position
+ * @param dsc           pointer to a vector graphic descriptor
+ * @param tx            the amount of translate in x direction
+ * @param tx            the amount of translate in y direction
+ */
+void lv_vector_dsc_translate(lv_vector_dsc_t * dsc, float tx, float ty);
+
+/**
+ * Change the skew factor of current transformation matrix
+ * @param dsc           pointer to a vector graphic descriptor
+ * @param skew_x        the skew factor for x direction
+ * @param skew_y        the skew factor for y direction
+ */
+void lv_vector_dsc_skew(lv_vector_dsc_t * dsc, float skew_x, float skew_y);
+
+/**
+ * Add a graphic path to the draw list
+ * @param dsc           pointer to a vector graphic descriptor
+ * @param path          pointer to a path
+ */
+void lv_vector_dsc_add_path(lv_vector_dsc_t * dsc, const lv_vector_path_t * path);
+
+/**
+ * Clear a rectangle area use current fill color
+ * @param dsc           pointer to a vector graphic descriptor
+ * @param rect          the area to clear in the buffer
+ */
+void lv_clear_rect(lv_vector_dsc_t * dsc, const lv_area_t * rect);
```

Comment:
This name is too generic. What about `lv_draw_vector_clear_rect()`

Reply:
Clear uses the current fill color, which is a separate operation so that the gpu's quick fill feature can be used later. As for the name, the original lv_vector_clear seems more reasonable to me, as it has the same prefix as the other functions

Reply:
What about `lv_vector_clear_rect()` or `lv_vector_clear_area` (to avoid abbreviation)?

Reply:
> What about `lv_vector_clear_rect()` or `lv_vector_clear_area` (to avoid abbreviation)?

 Putting it this way,  `lv_vector_clear_area`  maybe be better.

Reply:
Okay, please update it to `lv_vector_clear_area` and let's merge the feature.

We can add smaller PRs later if needed.

Reply:
> Okay, please update it to `lv_vector_clear_area` and let's merge the feature.
> 
> We can add smaller PRs later if needed.

ok